### PR TITLE
 Added tests for dvr router type change

### PR DESCRIPTION
**Pull request by TregubovYury**
```
Added tests for QA-311 QA-312 QA-313 QA-315
                QA-287 QA-288 QA-438 QA-439
Including applied commits for the pull #86 and #92
```

### CODE REVIEW

File: mos_tests/neutron/python_tests/test_dvr.py [hunk 1]
```diff
@@ -273,3 +277,343 @@ def test_routing_after_reset_computes(self, prepare_openstack, env_name):
...
+
+    def get_snat_controller(self, host_name, router_id):
+        snat_controller = None
+        controllers = self.env.get_nodes_by_role('controller')
+        for node in controllers:
+            if node.data['fqdn'] == host_name:
+                with self.env.get_ssh_to_node(node.data['ip']) as remote:
+                    cmd = 'ip netns | grep [s]nat-{}'.format(router_id)
+                    result = remote.execute(cmd)
+                    if any('snat' in s for s in result['stdout']):
```
Comment (gdyuldin) [inline mos_tests/neutron/python_tests/test_dvr.py]:
```
if result['exit_code'] == 0:
    snat_controller = node
    break
```

Comment (TregubovYury) [inline mos_tests/neutron/python_tests/test_dvr.py]: done

File: mos_tests/neutron/python_tests/test_dvr.py [hunk 2]
```diff
@@ -387,3 +391,343 @@ def test_routing_east_west(self, prepare_openstack):
...
+        tenant_name = self.os_conn.keystone.project_name
+        neutron = neutronclient.Client(username=username,
+                                       password=userpass,
+                                       tenant_name=tenant_name,
+                                       auth_url=auth_url,
+                                       ca_cert=path_to_cert)
+
+        # Try to create router with explicit distributed True value
+        # by user with memeber role but in admin tenant
+        with pytest.raises(NeutronClientException) as e:
```
Comment (TimurNurlygayanov) [inline mos_tests/neutron/python_tests/test_dvr.py:716]: Why we need it here?

File: mos_tests/neutron/python_tests/test_dvr.py [hunk 3]
```diff
@@ -387,3 +391,343 @@ def test_routing_east_west(self, prepare_openstack):
...
+            router = {'name': 'router01', 'distributed': True}
+            router_id = neutron.create_router(
+                            {'router': router})['router']['id']
+        allowed_msg = 'disallowed by policy'
+        err_msg = 'Failed to create the router, exception: {}'.format(e)
+        assert allowed_msg in str(e.value), err_msg
+
+        # Try to create router with explicit distributed False value
+        # by user with memeber role but in admin tenant
+        with pytest.raises(NeutronClientException) as e:
```
Comment (TimurNurlygayanov) [inline mos_tests/neutron/python_tests/test_dvr.py:731]: it is not clear how we use 

pytest.raises(NeutronClientException) as e

here

**Pull request merged**